collect_app/src/main/java/org/odk/collect/android/openrosa/OpenRosaResponseParserImpl.kt

```diff
@@ -0,0 +1,220 @@
+                        if (hash != null && (hash.length == 0 || !hash.startsWith("md5:"))) {
+                            hash = null
+                        } else {
+                            hash = hash.substring("md5:".length)
+                        }
+                    }
+                }
+            }
+
+            if (formId == null || downloadUrl == null || formName == null) {
+                formList.clear()
+                return null
+            }
+
+            val formListItem =
+                FormListItem(downloadUrl, formId, version!!, hash, formName, manifestUrl)
+            formList.add(formListItem)
+        }
+
+        return formList
+    }
+
+    override fun parseManifest(document: Document): List<MediaFile>? {
+        // Attempt OpenRosa 1.0 parsing
+        val manifestElement = document.rootElement
+        if (manifestElement.name != "manifest") {
+            return null
+        }
+
+        if (!isXformsManifestNamespacedElement(manifestElement)) {
+            return null
+        }
+
+        val elements = manifestElement.childCount
+        val files: MutableList<MediaFile> = ArrayList()
+        for (i in 0 until elements) {
+            if (manifestElement.getType(i) != Element.ELEMENT) {
+                // e.g., whitespace (text)
+                continue
+            }
+
+            val mediaFileElement = manifestElement.getElement(i)
+            if (!isXformsManifestNamespacedElement(mediaFileElement)) {
+                // someone else's extension?
+                continue
+            }
+
+            val name = mediaFileElement.name
+            if (name.equals("mediaFile", ignoreCase = true)) {
+                var filename: String? = null
+                var hash: String? = null
+                var downloadUrl: String? = null
+                // don't process descriptionUrl
+                val childCount = mediaFileElement.childCount
+                for (j in 0 until childCount) {
+                    if (mediaFileElement.getType(j) != Element.ELEMENT) {
+                        // e.g., whitespace (text)
+                        continue
+                    }
+
+                    val child = mediaFileElement.getElement(j)
+                    if (!isXformsManifestNamespacedElement(child)) {
+                        // someone else's extension?
+                        continue
+                    }
+
+                    when (child.name) {
+                        "filename" -> {
+                            filename = XFormParser.getXMLText(child, true)
+                            if (filename != null && filename.length == 0) {
+                                filename = null
+                            }
+                        }
+                        "hash" -> {
+                            hash = XFormParser.getXMLText(child, true)
+                            if (hash != null && hash.length == 0) {
```

Comment:
Don't you need to check if it starts with `md5:` here too?

Reply:
Ah you want to improve it later https://github.com/getodk/collect/pull/4911/files#diff-97ea04c120d7632ab55eb3218d1ec61bf0c45056bc43f9ca6c6573feac4250cbR57?

Reply:
Yeah exactly!